vscripts/confogl.nut

```diff
@@ -2,65 +2,308 @@
 MutationOptions <-
 {
   cm_ProhibitBosses = 0
- 	cm_AllowPillConversion = 0
+  cm_AllowPillConversion = 0
 }
 
-
 local pluginScopes = []
-Hooks <- {
-  function AllowBash(basher, bashee) {
-    // Default to allow the bash.
-    local retVal = true;
-	  for ( local idx = pluginScopes.len(); idx >= 0; --idx ) {
-      local nextPlugin = pluginScopes[idx];
-      local fn = pluginScopes[idx].AllowBash;
-      if ( typeof (fn) == "function") {
-        local res = fn(basher, bashee)
-        // If any plugin returns false, we return false.
-        if(res == false) {
-          return false;
-        }
-      }
+
+// int AllowBash(handle basher, handle bashee): called whenever melee bash is used.
+// Returns these values:
+// ALLOW_BASH_ALL - normal melee behavior
+// ALLOW_BASH_NONE - do nothing at all
+// ALLOW_BASH_PUSHONLY - applies physics push but deals no damage (including prevention of insta-kill ambush behavior)
+function AllowBash(basher, bashee) {
+  Msg(format("AllowBash(%s, %s)\n", basher.tostring(), bashee.tostring()));
+  // Default to allow the bash.
+  local retVal = ALLOW_BASH_ALL;
+  for ( local idx = pluginScopes.len() - 1; idx >= 0; --idx ) {
+    local nextPlugin = pluginScopes[idx];
+    if ("AllowBash" in pluginScopes[idx]) {
+	  local fn = pluginScopes[idx].AllowBash;
+	  if ( typeof (fn) == "function" ) {
+	    local res = fn(basher, bashee)
+	    // If any plugin returns ALLOW_BASH_NONE, we return ALLOW_BASH_NONE.
+	    if(res == ALLOW_BASH_NONE) {
+		  return ALLOW_BASH_NONE;
+	    }
+	    // If any plugin returns ALLOW_BASH_PUSHONLY, we return ALLOW_BASH_PUSHONLY.
+	    if(res == ALLOW_BASH_PUSHONLY) {
+		  return ALLOW_BASH_PUSHONLY;
```

Comment:
I take it we want `ALLOW_BASH_NONE` to be the highest precedent, and then PUSHONLY after that. In which case, we probably just want to set the retVal here: 
```suggestion
		  retVal =  ALLOW_BASH_PUSHONLY;
```

Reply:
Implemented: https://github.com/Icy-Inferno/confogl-mutation/commit/f38354693dcdf832d2afd46c73a08962ff786002